x-pack/plugins/lens/public/embeddable/embeddable.tsx

```diff
@@ -185,6 +185,7 @@ interface LensBaseEmbeddableInput extends EmbeddableInput {
     data: Simplify<LensTableRowContextMenuEvent['data'] & PreventableEvent>
   ) => void;
   abortController?: AbortController;
+  customBadgeMessages?: (userMessages: UserMessage[]) => UserMessage[];
```

Comment:
naming things is not my best skill, but I would prefer to have something here that would hint to the consumer what this prop does.
What about:
```suggestion
  onBeforeBadgesRendering?: (userMessages: UserMessage[]) => UserMessage[];
```

I'm open to other names here too.

Reply:
I am not convinced by Marco's choice on the name here. It's not really that relevant that:
- this callback would execute before badges rendering, more the whole user messages rendering
- using the word `badges` is very specific for no reason unless I don't understand something. 

This function can do more than just operate on badges, basically overriding the whole user messages content if needed. I'd go with something more generic, like `transformUserMessages`, `userMessagesCustomTransform` or even `onBeforeUserMessagesRendering`.
But we can talk inside the Lens team about so after some testing I'll approve.